metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -565,10 +602,9 @@ case class ScalafixProvider(
   private def getRuleClassLoader(
       scalfixRulesKey: ScalafixRulesClasspathKey,
       scalafixClassLoader: ClassLoader,
-  ): Future[URLClassLoader] = {
-    rulesClassloaderCache.get(scalfixRulesKey) match {
```

Comment:
Potentially, this could get invoked from two different processes and return api twice, first being overridden by the second and then the first one's classloader would still be used for the tool classloader, which could make it inconpatible.